### PR TITLE
Refactor Queue for namespace replication

### PR DESCRIPTION
**Pull request by yiminc**
**What changed?**
Refactored Queue (for namespace replication).

**Why?**
Clear layer separation for persistence

**How did you test it?**
Unit tests and integration tests

**Potential risks**
Low

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/persistence/cassandra/cassandraQueue.go [hunk 1]
```diff
@@ -333,69 +313,33 @@ func (q *cassandraQueue) getQueueMetadata(
...
 	queueType persistence.QueueType,
 ) error {
 
-	blob, err := serialization.QueueMetadataToBlob(&persistencespb.QueueMetadata{
-		ClusterAckLevels: metadata.clusterAckLevels,
-	})
-	if err != nil {
-		return err
-	}
+	// TODO: remove this once cluster_ack_level is removed from DB
```
Comment (wxing1292) [inline common/persistence/cassandra/cassandraQueue.go:321]: i will take care of this schema deprecation.

File: common/persistence/namespaceReplicationQueue.go [hunk 2]
```diff
@@ -223,25 +324,15 @@ func (q *namespaceReplicationQueueImpl) GetMessagesFromDLQ(
...
-	return nil
+	return q.updateAckLevelWithRetry(lastProcessedMessageID, localNamespaceReplicationCluster, true)
 }
 
 func (q *namespaceReplicationQueueImpl) GetDLQAckLevel() (int64, error) {
-	dlqMetadata, err := q.queue.GetDLQAckLevels()
+	metadata, err := q.queue.GetDLQAckLevels()
+	if err != nil {
+		return EmptyQueueMessageID, err
+	}
```
Comment (wxing1292) [inline common/persistence/namespaceReplicationQueue.go:334]:
```go
switch err.(type) {
case *serviceerror.NotFound:
    return EmptyQueueMessageID, nil
default:
    return nil, err
}

Comment (wxing1292) [reply common/persistence/namespaceReplicationQueue.go:334]: and make queue impl return serviceerror.NotFound when there is no record?

**Review by wxing1292 — APPROVED** [review]

File: common/persistence/namespaceReplicationQueue.go (hunk 2 above)
Comment (yiminc) [reply common/persistence/namespaceReplicationQueue.go:334]: This method returns a int64, cannot use nil. The error from store layer should already handle the case.

**Pull request merged**